application/src/main/java/org/opentripplanner/model/plan/Emission.java

```diff
@@ -0,0 +1,75 @@
+package org.opentripplanner.model.plan;
+
+import java.io.Serializable;
+import java.util.Objects;
+import org.opentripplanner.framework.model.Gram;
+import org.opentripplanner.utils.lang.Sandbox;
+import org.opentripplanner.utils.tostring.ToStringBuilder;
+
+/**
+ * Represents the emission of a journey. Each type of emissions has its own field and unit.
+ */
+@Sandbox
+public final class Emission implements Serializable {
+
+  public static final Emission ZERO = new Emission(Gram.ZERO);
+
+  private final Gram co2;
+
+  private Emission(Gram co2) {
+    this.co2 = Objects.requireNonNull(co2);
+  }
+
+  public static Emission of(Gram co2) {
+    return new Emission(co2);
+  }
+
+  /** Convinience factory method for creating test data */
+  public static Emission co2_g(double co2_g) {
+    return new Emission(Gram.of(co2_g));
+  }
```

Comment:
You use this outside of tests too.

Reply:
Not any more, see PR #6628. The "natural laws" of writing good encapsulated code from A - Z, make sure something is mapped as input to the system and stay encapsulated. Since, the mappers usually do not use a factory method of this kind, usage of these indicate that full encapsulation is not reached. Hence; the comment from me in the first place - I did not expect it to be used in the main code. After a full cleanup, using the type in the configuration as well - the usage disappeared.